### ISSUE TITLE
White page when upgrading from 1.0.1 to 1.0.3 (and up)

### ISSUE THREAD

**Issue by man-andy** [tes/compoxure#73]
I did an update from 1.0.1 to master and when I went to the site, I got a white page.
I also tried version 1.0.3 and it will also result in a white page.

The page gives a status 200, but the response is empty. 
I do not see any error that is causing this problem. I only see this in my terminal:
```
127.0.0.1 - - [20/Jun/2016:13:08:02 +0000] "GET /page/ HTTP/1.1" 200 - "http://www.mylocalsite.com" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.84 Safari/537.36 FirePHP/4Chrome"
127.0.0.1 - - [20/Jun/2016:13:08:02 +0000] "GET /favicon.ico HTTP/1.1" 200 - "http://www.mylocalsite.com" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.84 Safari/537.36 FirePHP/4Chrome"
```

> (changed my url to http://www.mylocalsite.com)

**Comment by cliftonc** [comment]
Doesn't sound good - are you able to share the config or the page so we can attempt to recreate?

**Comment by man-andy** [comment]
Yeah, sure:

config.json
``` json
{
    "parameters": {
        "servers": {
            "matchengine": "http://www.google.nl"
        }
    },
    "backend": [
        {
            "pattern": ".*",
            "timeout": "1000",
            "parserTimeout": "20000",
            "target": "http://localhost:4000",
            "dontPassUrl": false,
            "quietFailure": false,
            "noCache": true
        }
    ],
    "statusCodeHandlers":{
        "403":{
            "fn":"handleUnauthorised"
        },
        "401":{
            "fn":"handleUnauthorised"
        }
    }
}
```

index.js
``` js
'use strict';

var cx = require('compoxure');
var config = require('./config.json');
var connect = require('connect');
var cookieParser = require('cookie-parser');
var query = require('connect-query');
var morgan = require('morgan');

function createEventHandler() {
    return {
        logger: function(level, message) {
            if(process.env.logging !== 'false') { console.log('LOG ' + level + ': ' + message); }
        },
        stats: function(type, key, value) {
            if(process.env.logging !== 'false') { console.log('STAT ' + type + ' for ' + key + ' | ' + value); }
        }
    };
}
var cxEventHandler = createEventHandler();

config.functions = {
    'handleUnauthorised': function(req, res, variables) {
        if (res.headersSent) { return; } // too late to redirect
        res.writeHead(302, {'Location': '/no-access/?returnurl='+variables['url:path:encoded']});
        res.end();
    }
};
config.environment = 'development';
config.minified = false;

var compoxureMiddleware = cx(config, cxEventHandler);

var server = connect();
server.use(cookieParser());
server.use(query());
if(process.env.logging !== 'false') { server.use(morgan('combined')); }
server.use(compoxureMiddleware);

server.listen(5000, 'localhost', function() {
    console.log('server on http://localhost:5000');
});
```

package.json
```json
{
  "name": "test",
  "version": "0.1.0",
  "description": "test",
  "main": "index.js",
  "scripts": {
    "lint": "jshint ."
  },
  "dependencies": {
    "connect": "^3.3.5",
    "cookie-parser": "^1.3.2",
    "connect-query": "^0.2.0",
    "morgan": "^1.2.1",
    "compoxure": "^1.0.5"
  },
  "devDependencies": {
    "jshint": "^2.9.2",
    "precommit-hook": "^1.0.7"
  },
  "config": {
    "blanket": {
      "pattern": "src"
    }
  },
  "pre-commit": [
    "lint"
  ]
}
```

**Comment by cliftonc** [comment]
Can you try adding the following to the config:

```
 "fragmentPasses": -1
```

This disables the multi-pass rendering that was introduced around 1.0.0.

**Comment by man-andy** [comment]
No joy...

Perhaps there is a difference for nodejs version support between 1.0.1 and 1.0.3?

**Comment by cliftonc** [comment]
Not that I know of, what version of node are you running?  What is the HTML you're responding with in the backend?

**Comment by man-andy** [comment]
I am running an old version of nodejs, because I use the Ubuntu repository. v0.12.8
However, if the version support didn't change between the versions, this doesn't have to be the problem.

I have tried difference HTML.
The default Symfony index page, Google.com homepage and also a html page containing only a "hello world".
All 3 pages are working correctly in 1.0.1, but not in 1.0.3 - 1.0.5

**Comment by cliftonc** [comment]
Ok, I'll keep digging.

**Comment by datenpate** [comment]
Same is happening for me. Releases from 1.0.3 are producing a blank page. 1.0.1 is still working.
```
# node -v
v6.2.2
```

**Comment by datenpate** [comment]
@cliftonc I also did some digging. Starting with v1.0.3 ``middleware/backend.js:selectBackend()`` is not called anymore on request. Maybe there is something broken in https://github.com/tes/compoxure/commit/5c49cb0aeeea44dc094f0a8f54237e48fc8b9a22

**Comment by RonRademaker** [comment]
@cliftonc I found the problem for our situation and fixed it in PR #75, does this solution work for you too?

**Comment by cliftonc** [comment]
Lgtm but am on holiday so will need someone else to merge, @gabceb ?
